scripts/4.Covariates.py

```diff
@@ -0,0 +1,156 @@
+
+# coding: utf-8
+
+# # Incorporating covariate information into classifiers
+
+# Daniel has shown that the sample data possesses covariates in <a href='https://github.com/cognoma/cancer-data/blob/master/3.explore-mutations.ipynb'>this notebook</a>. These may provide a spurious signal that a classifier accommodates for, and could confound attempts to pick out the actual signal we desire. This notebook will create a file with encoded information on these covariates that can be used as additional data to train on.
+
+# In[1]:
+
+import os
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+get_ipython().magic('matplotlib inline')
+
+
+# Let's peak at the sample data.
+
+# In[2]:
+
+path = os.path.join('download', 'samples.tsv')
+samples_df = pd.read_table(path, index_col=0)
+samples_df.head(10)
+
+
+# Let's get a sense of how much missing data there is.
+
+# In[3]:
+
+print('Total number of samples: %d' % len(samples_df))
+print('Number of nulls in each column:')
+samples_df.isnull().sum(axis=0)
+
+
+# Now count the number of types of each column. We'll use Python's set, and therefore must exclude null values.
+
+# In[4]:
+
+categorical_variables = ['sample_type', 'disease', 'organ_of_origin', 'gender','dead', 'recurred']
+for variable in categorical_variables:
+    not_null = samples_df[variable].notnull()
+    count = len(set(samples_df.loc[not_null, variable]))
+    print('Number of types in category %s: %d' % (variable, count))
+
+
+# There are no missing values from <code>sample_types</code> and only one possible value, so it is redundant and we'll exclude it.
+
+# In[5]:
+
+del categorical_variables[categorical_variables.index('sample_type')]
+print(categorical_variables)
+
+
+# Inspecting the head of the samples DataFrame above, we see that two columns, namely <code>dead</code> and <code>recurred</code>, need some attention. We're going to encode categorical variables using panda's get_dummies. Since these columns are indicated by a 1 or 0, this will become the column header when encoded, as below.
+
+# In[6]:
+
+pd.get_dummies(samples_df.dead).head(10)
+
+
+# Note that null values are encoded as (0,0) pairs.
+# 
+# Let's rename the values in each of these so that they more accurately reflect the underlying variable.
+
+# In[7]:
+
+mortality_map = pd.Series(['alive', 'dead'], index=[0.0, 1.0])
+samples_df.dead = samples_df.dead.map(mortality_map)
+samples_df.dead.head(10)
+
+
+# In[8]:
+
+recurred_map = pd.Series(['has_not_recurred', 'has_recurred'], index=[0.0, 1.0])
```

Comment:
Not sure if you understood my previous comment here. The suggestion is to avoid any value mapping here. Then once you create `covariates_df` you can use

```python
renamer = {
    'dead_0': 'alive',
    'dead_1': 'dead',
   # keep going for everything you want to rename
}
covariates_df.rename(columns=renamer, inplace=True)
```

As an aside, I think it's more succinct to map using a dictionary rather than creating a series.